test/resources/OAuth.spec.js

```diff
@@ -0,0 +1,177 @@
+'use strict';
+
+var stripe = require('../../testUtils').getSpyableStripe();
+
+var expect = require('chai').expect;
+
+describe('OAuth', function() {
+  describe('authorize', function() {
+    describe('when a default client_id is not set', function() {
+      beforeEach(function() {
+        stripe.setClientId('');
+      });
+
+      describe('with an explicitly provided client_id', function() {
+        it('Generates the correct URL', function() {
+          var url = stripe.oAuth.authorizeUrl({client_id: '123abc'});
+
+          var expectedUrl = 'https://connect.stripe.com/oauth/authorize?client_id=123abc&response_type=code&scope=read_write';
+
+          expect(url).to.equal(expectedUrl);
+        });
+      })
+    });
+
+    describe('when a default client_id is set', function() {
+      beforeEach(function() {
+        stripe.setClientId('default_client_id');
+      });
+
+      describe('when required parameters are not provided', function() {
+        it('Generates the correct URL', function() {
+          var url = stripe.oAuth.authorizeUrl();
+
+          var expectedUrl = 'https://connect.stripe.com/oauth/authorize?response_type=code&client_id=default_client_id&scope=read_write';
+
+          expect(url).to.equal(expectedUrl);
+        });
+
+        it('Generates the correct URL with state provided', function() {
+          var url = stripe.oAuth.authorizeUrl({state: 'some_state'});
+
+          var expectedUrl = 'https://connect.stripe.com/oauth/authorize?state=some_state&response_type=code&client_id=default_client_id&scope=read_write';
+
+          expect(url).to.equal(expectedUrl);
```

Comment:
These tests are a bit hard to read/verify, and I can imagine someone in the future "fixing" the tests in a way that hides a bug. 

Could we get a bit more granular? I'm thinking something like this:

```js
var URL = require('url');
var qs = require('qs');

// in this test
var state = qs.parse(URL.parse(url).query).state;
expect(state).to.equal('some_state');

// in other tests
var pathname = URL.parse(url).pathname;
expect(pathname).to.equal('/express/oauth/authorize');
```